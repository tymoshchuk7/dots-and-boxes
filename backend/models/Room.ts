import { DataTypes } from 'sequelize';
import db from '../db';
import { RoomInstance, RoomStatus } from '../types/RoomTypes';

export default function defineRoom() {
  const RoomModel = db.define<RoomInstance>('room', {
    id: {
      type: DataTypes.UUID,
      allowNull: false,
      primaryKey: true,
      defaultValue: DataTypes.UUIDV4,
    },
    name: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
    },
    status: {
      type: DataTypes.ENUM({ values: Object.values(RoomStatus) }),
      allowNull: false,
      defaultValue: RoomStatus.NEW_ROOM,
    },
    width: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    height: {
      type: DataTypes.INTEGER,
      allowNull: false,
    },
    sticks: {
      type: DataTypes.JSONB,
    },
    boxes: {
      type: DataTypes.JSONB,
    },
    move_timestamp: {
      type: DataTypes.DATE,
    },
  }, {
    underscored: true,
  });
  return RoomModel;
}
