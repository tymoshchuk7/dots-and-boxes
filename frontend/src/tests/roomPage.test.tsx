import React from 'react';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Room from '../views/roomPage';

describe('Room page tests', () => {
  const middlewares = [thunk];
  let testStore;
  let component;
  const mockStore = configureStore(middlewares);
  beforeEach(() => {
    testStore = mockStore({
      verifyUser: {
        id: '123',
      },
      score: {
        123: '1',
        456: '2',
        789: '3',
      },
      game: {
        playerList: ['123', '456'],
        sticks: [
          [{}, {}],
          [{}, {}, {}],
          [{}, {}],
          [{}, {}, {}],
          [{}, {}],
        ],
        boxes: [
          [{ byPlayer: null }, { byPlayer: '123' }],
          [{ byPlayer: '123' }, { byPlayer: '456' }],
        ],
        status: 'Active',
        roomPlayers: [{ id: '123' }, { id: '456' }, { id: '789' }],
      },
      player: {
        id: '123',
      },
    });
    component = renderer.create(
      <Provider store={testStore}>
        <BrowserRouter>
          <Room />
        </BrowserRouter>
      </Provider>,
    );
  });

  test('snapshoting the render of component', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
  test('check call of function on stick chose', () => {
    renderer.act(() => {
      const horizontalSticks = component.root.findAllByProps({ className: 'horiz' });
      const verticalSticks = component.root.findAllByProps({ className: 'vert' });
      expect(horizontalSticks).toHaveLength(3);
      expect(verticalSticks).toHaveLength(6);
    });
  });
});
