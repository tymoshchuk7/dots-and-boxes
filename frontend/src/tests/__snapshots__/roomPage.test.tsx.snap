// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room page tests snapshoting the render of component 1`] = `
<div
  className="room"
>
  <div
    className="d-flex justify-content-center pt-4"
  >
    <div
      className="d-flex flex-column"
    >
      <div
        className="playercard-wrapper "
      >
        <div
          className="playercard"
        >
          <div
            className="text-center playercard__header"
            style={
              Object {
                "background": "radial-gradient(black,#ff0000)",
                "color": "#ffffff",
              }
            }
          >
              points
          </div>
          <div
            className="playercard-border playercard-name pl-3"
          >
             
            (me)
          </div>
          <div
            className="d-flex"
          >
            <div
              className="playercard-border"
            >
              <img
                alt="userPhoto"
                src="http://localhost:7000/undefined"
              />
            </div>
            <div
              className="playercard-border"
            >
              <button
                className="btn btn-secondary btn-sm"
                disabled={false}
                type="button"
              >
                Action
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="playercard-wrapper "
      >
        <div
          className="playercard"
        >
          <div
            className="text-center playercard__header"
            style={
              Object {
                "background": "radial-gradient(black,#ff0000)",
                "color": "#ffffff",
              }
            }
          >
              points
          </div>
          <div
            className="playercard-border playercard-name pl-3"
          >
             
            
          </div>
          <div
            className="d-flex"
          >
            <div
              className="playercard-border"
            >
              <img
                alt="userPhoto"
                src="http://localhost:7000/undefined"
              />
            </div>
            <div
              className="playercard-border"
            >
              <button
                className="btn btn-secondary btn-sm"
                disabled={false}
                type="button"
              >
                Action
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="d-flex flex-column"
    >
      <div
        className="players-mobile"
      >
        <div
          className="d-flex playercad-border--mobile align-items-center justify-content-around p-1 "
          style={
            Object {
              "backgroundImage": "url(/bar.jpg)",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "100% 100%",
            }
          }
        >
          <img
            alt="userPhoto"
            src="http://localhost:7000/undefined"
          />
          <div>
             
            (me)
          </div>
          <div>
              points
          </div>
          <div
            style={
              Object {
                "backgroundColor": "#ff0000",
                "border": "1px solid black",
                "height": "15px",
                "width": "15px",
              }
            }
          />
        </div>
        <div
          className="d-flex playercad-border--mobile align-items-center justify-content-around p-1 "
          style={
            Object {
              "backgroundImage": "url(/bar.jpg)",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "100% 100%",
            }
          }
        >
          <img
            alt="userPhoto"
            src="http://localhost:7000/undefined"
          />
          <div>
             
            
          </div>
          <div>
              points
          </div>
          <div
            style={
              Object {
                "backgroundColor": "#ff0000",
                "border": "1px solid black",
                "height": "15px",
                "width": "15px",
              }
            }
          />
        </div>
        <div
          className="d-flex playercad-border--mobile align-items-center justify-content-around p-1 "
          style={
            Object {
              "backgroundImage": "url(/bar.jpg)",
              "backgroundRepeat": "no-repeat",
              "backgroundSize": "100% 100%",
            }
          }
        >
          <img
            alt="userPhoto"
            src="http://localhost:7000/undefined"
          />
          <div>
             
            
          </div>
          <div>
              points
          </div>
          <div
            style={
              Object {
                "backgroundColor": "#ff0000",
                "border": "1px solid black",
                "height": "15px",
                "width": "15px",
              }
            }
          />
        </div>
      </div>
      <div
        className="field-wrapper"
      >
        <div
          className="d-flex justify-content-center board"
        >
          <div
            className="d-flex flex-column gameboard"
          >
            <div
              className="d-flex flex-row"
            >
              <div
                className="horiz-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="horiz"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "width": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
              <div
                className="horiz-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="horiz-last"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "width": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
            </div>
            <div
              className="d-flex flex-row"
            >
              <div
                className="d-flex flex-row"
              >
                <div
                  className="vert-wrapper"
                >
                  <button
                    aria-label="click stick"
                    className="vert"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#ff0000",
                        "height": "15vw",
                      }
                    }
                    type="button"
                  />
                </div>
                <div
                  className="box d-flex justify-content-center align-items-center"
                  style={
                    Object {
                      "backgroundPosition": "center",
                      "backgroundRepeat": "no-repeat",
                      "backgroundSize": "60%",
                      "height": "15vw",
                      "width": "15vw",
                    }
                  }
                >
                  <p
                    className="m-0 p-0"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "36px",
                        "position": "relative",
                      }
                    }
                  >
                    <span
                      className="box__letter"
                      style={
                        Object {
                          "fontSize": "7.5vw",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
              <div
                className="d-flex flex-row"
              >
                <div
                  className="vert-wrapper"
                >
                  <button
                    aria-label="click stick"
                    className="vert"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#ff0000",
                        "height": "15vw",
                      }
                    }
                    type="button"
                  />
                </div>
                <div
                  className="box d-flex justify-content-center align-items-center"
                  style={
                    Object {
                      "backgroundPosition": "center",
                      "backgroundRepeat": "no-repeat",
                      "backgroundSize": "60%",
                      "height": "15vw",
                      "width": "15vw",
                    }
                  }
                >
                  <p
                    className="m-0 p-0"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "36px",
                        "position": "relative",
                      }
                    }
                  >
                    <span
                      className="box__letter"
                      style={
                        Object {
                          "fontSize": "7.5vw",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
              <div
                className="vert-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="vert"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "height": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
            </div>
            <div
              className="d-flex flex-row"
            >
              <div
                className="horiz-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="horiz"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "width": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
              <div
                className="horiz-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="horiz-last"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "width": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
            </div>
            <div
              className="d-flex flex-row"
            >
              <div
                className="d-flex flex-row"
              >
                <div
                  className="vert-wrapper"
                >
                  <button
                    aria-label="click stick"
                    className="vert"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#ff0000",
                        "height": "15vw",
                      }
                    }
                    type="button"
                  />
                </div>
                <div
                  className="box d-flex justify-content-center align-items-center"
                  style={
                    Object {
                      "backgroundPosition": "center",
                      "backgroundRepeat": "no-repeat",
                      "backgroundSize": "60%",
                      "height": "15vw",
                      "width": "15vw",
                    }
                  }
                >
                  <p
                    className="m-0 p-0"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "36px",
                        "position": "relative",
                      }
                    }
                  >
                    <span
                      className="box__letter"
                      style={
                        Object {
                          "fontSize": "7.5vw",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
              <div
                className="d-flex flex-row"
              >
                <div
                  className="vert-wrapper"
                >
                  <button
                    aria-label="click stick"
                    className="vert"
                    onClick={[Function]}
                    onMouseOut={[Function]}
                    onMouseOver={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#ff0000",
                        "height": "15vw",
                      }
                    }
                    type="button"
                  />
                </div>
                <div
                  className="box d-flex justify-content-center align-items-center"
                  style={
                    Object {
                      "backgroundPosition": "center",
                      "backgroundRepeat": "no-repeat",
                      "backgroundSize": "60%",
                      "height": "15vw",
                      "width": "15vw",
                    }
                  }
                >
                  <p
                    className="m-0 p-0"
                    style={
                      Object {
                        "color": "#fff",
                        "fontSize": "36px",
                        "position": "relative",
                      }
                    }
                  >
                    <span
                      className="box__letter"
                      style={
                        Object {
                          "fontSize": "7.5vw",
                        }
                      }
                    />
                  </p>
                </div>
              </div>
              <div
                className="vert-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="vert"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "height": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
            </div>
            <div
              className="d-flex flex-row"
            >
              <div
                className="horiz-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="horiz"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "width": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
              <div
                className="horiz-wrapper"
              >
                <button
                  aria-label="click stick"
                  className="horiz-last"
                  onClick={[Function]}
                  onMouseOut={[Function]}
                  onMouseOver={[Function]}
                  style={
                    Object {
                      "backgroundColor": "#ff0000",
                      "width": "15vw",
                    }
                  }
                  type="button"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      
      <div
        className="info text-center p-2"
      >
        Time left:
         
        0
      </div>
      
      
    </div>
    <div
      className="d-flex flex-column"
    >
      <div
        className="playercard-wrapper "
      >
        <div
          className="playercard"
        >
          <div
            className="text-center playercard__header"
            style={
              Object {
                "background": "radial-gradient(black,#ff0000)",
                "color": "#ffffff",
              }
            }
          >
              points
          </div>
          <div
            className="playercard-border playercard-name pl-3"
          >
             
            
          </div>
          <div
            className="d-flex"
          >
            <div
              className="playercard-border"
            >
              <img
                alt="userPhoto"
                src="http://localhost:7000/undefined"
              />
            </div>
            <div
              className="playercard-border"
            >
              <button
                className="btn btn-secondary btn-sm"
                disabled={false}
                type="button"
              >
                Action
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
